composer.json

```diff
@@ -23,11 +23,9 @@
     "require": {
         "php": "^7.1",
         "cocur/slugify": "^2.0 || ^3.0",
-        "sonata-project/admin-bundle": "^3.31",
         "sonata-project/core-bundle": "^3.14",
         "sonata-project/datagrid-bundle": "^2.3",
         "sonata-project/doctrine-extensions": "^1.5.1",
-        "sonata-project/doctrine-orm-admin-bundle": "^3.4",
```

Comment:
Is this why the coverage dropped so much?

Reply:
Good catch. I'll have a look at this

Reply:
There is no usage for any `DoctrineORMAdminBundle` class 🤔

Reply:
Maybe there is usage for a service?

Reply:
Coverage fell for ORM entities… they might no longer be loaded in a functional test when that bundle is missing?

Reply:
I'm a bit confused. We don't have any asserts for these classes. Why is the coverage decreased

Reply:
You can try throwing in the constructor of these classes and see if there is a functional test that actually covers them?

Reply:
Sorry, I don't get it :(

Reply:
Ok let's take `BaseCategory` as an example. If I throw in the code of this class, I get this:

```
1) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetCategoriesWithMultipleRootsInContext
Exception: 

/home/greg/dev/SonataClassificationBundle/src/Entity/BaseCategory.php:23
/home/greg/dev/SonataClassificationBundle/src/Entity/CategoryManager.php:286
/home/greg/dev/SonataClassificationBundle/src/Entity/CategoryManager.php:197
/home/greg/dev/SonataClassificationBundle/tests/Entity/CategoryManagerTest.php:108

2) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetRootCategoryWithChildren
Exception: 

/home/greg/dev/SonataClassificationBundle/src/Entity/BaseCategory.php:23
/home/greg/dev/SonataClassificationBundle/src/Entity/CategoryManager.php:286
/home/greg/dev/SonataClassificationBundle/src/Entity/CategoryManager.php:95
/home/greg/dev/SonataClassificationBundle/tests/Entity/CategoryManagerTest.php:138

3) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetRootCategory
Exception: 

/home/greg/dev/SonataClassificationBundle/src/Entity/BaseCategory.php:23
/home/greg/dev/SonataClassificationBundle/src/Entity/CategoryManager.php:286
/home/greg/dev/SonataClassificationBundle/src/Entity/CategoryManager.php:117
/home/greg/dev/SonataClassificationBundle/tests/Entity/CategoryManagerTest.php:161

4) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetRootCategoriesForContext
Exception: 

/home/greg/dev/SonataClassificationBundle/src/Entity/BaseCategory.php:23
/home/greg/dev/SonataClassificationBundle/src/Entity/CategoryManager.php:286
/home/greg/dev/SonataClassificationBundle/src/Entity/CategoryManager.php:128
/home/greg/dev/SonataClassificationBundle/tests/Entity/CategoryManagerTest.php:192
```

Now if I try the same on `optional-admin`, I get an entirely different result:

```
1) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetPager
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

2) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetPagerWithEnabledCategories
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

3) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetPagerWithDisabledCategories
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

4) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetCategoriesWithMultipleRootsInContext
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

5) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetRootCategoryWithChildren
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

6) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetRootCategory
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

7) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetRootCategoriesForContext
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

8) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetRootCategories
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

9) Sonata\ClassificationBundle\Tests\Entity\CategoryManagerTest::testGetRootCategoriesSplitByContexts
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

10) Sonata\ClassificationBundle\Tests\Entity\CollectionManagerTest::testGetPager
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

11) Sonata\ClassificationBundle\Tests\Entity\CollectionManagerTest::testGetPagerWithEnabledCollections
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

12) Sonata\ClassificationBundle\Tests\Entity\CollectionManagerTest::testGetPagerWithDisabledCollections
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

13) Sonata\ClassificationBundle\Tests\Entity\ContextManagerTest::testGetPager
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

14) Sonata\ClassificationBundle\Tests\Entity\ContextManagerTest::testGetPagerWithEnabledContexts
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

15) Sonata\ClassificationBundle\Tests\Entity\ContextManagerTest::testGetPagerWithDisabledContexts
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

16) Sonata\ClassificationBundle\Tests\Entity\TagManagerTest::testGetPager
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

17) Sonata\ClassificationBundle\Tests\Entity\TagManagerTest::testGetPagerWithEnabledTags
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist

18) Sonata\ClassificationBundle\Tests\Entity\TagManagerTest::testGetPagerWithDisabledTags
Cannot stub or mock class or interface "Doctrine\ORM\AbstractQuery" which does not exist
```

Also, although I am in dev, I notice some dependencies are gone:

```
  - Removing sonata-project/doctrine-orm-admin-bundle (3.15.0)
  - Removing ocramius/package-versions (1.5.1)
  - Removing jdorn/sql-formatter (v1.2.17)
  - Removing doctrine/orm (v2.7.2)
  - Removing doctrine/doctrine-bundle (2.0.7)
  - Downgrading cocur/slugify (v4.0.0 => v3.2): Loading from cache
```

IMO you should move `sonata-project/doctrine-orm-admin-bundle` to the `require-dev` section instead of removing it entirely, that way these tests should work and the entities should be covered.

Reply:
Done